### DISCUSSION TITLE
proposal : let enums acting like vb.net acts

### DISCUSSION THREAD

**Discussion by rootflood** [dotnet/csharplang#3617] [Language Ideas]
sample valid code in vb.net:
```vb
'vb.net
    Enum myE
        E1 = 12
        E2 = 13
    End Enum

    Sub Main()
        Dim e As myE
        e = myE.E1
        e = e << 2
    End Sub
```
```c#
//C#
        enum myE
        {
            E1 = 12,
            E2 = 13
        }
        public static void Main()
        {
            myE e;
            e = myE.E1;
            e = e << 2;
        }
```
more details can be found here : https://github.com/dotnet/csharplang/issues/3572 , https://github.com/dotnet/csharplang/issues/3593

**Comment by theunrepentantgeek** [comment]
This has already been thoroughly discussed elsewhere - in C# doing bit shifts on the values of enumerations is only allowed during definition, as a convenience. Bit shifting on values outside the definition is prohibited because it's a fruitful source of bugs.

Please don't open duplicate issues just because you don't like the outcome of the conversation.

**Comment by rootflood** [comment]
They told me I can push proposal, so it's not duplicate, this is proposal, title are different and this focused to "enums" and not others. 
I think others may didn't like the outcome of that conversation so please don't judge me.
Thanks for your Comment.

**Comment by rootflood** [comment]
> @theunrepentantgeek >Bit shifting on values outside the definition is prohibited because it's a fruitful source of bugs.

Documents: The set of values that an enum type can take on is not limited by its enum members. [Enum values and operations](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/enums#enum-values-and-operations)

Valid Example:
```c#
    class Program
    {
        public enum a {a1=1}

        public static void Main()
        {
            // no error
            a a = a.a1 + 1000;
        }
    }
```

**Comment by CyrusNajmabadi** [comment]
The proposal is fine to have.   But please stop closing proposals that are effectively duplicates just to open the same ones again.  Thanks :)   I'll leave this open as i think it's totally fine to request this as a language change.
